### PR TITLE
Better edge-case handling of the layout snippet detection

### PR DESCRIPTION
**Pull request by lukasbestle**
### Refactoring (within 3.9 changes)

- Make `$snippet->parent` accessible
- Better edge-case handling of the layout snippet detection (follow-up to #4999)

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] Tests and checks all pass

### For review team

- [x] ~Add changes to release notes draft in Notion~
- [x] ~Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)~

### CODE REVIEW

File: src/Template/Template.php [hunk 1]
```diff
@@ -158,7 +158,10 @@ public function render(array $data = []): string
 		// has been omitted, take the buffer output as default
 		// slot and render the snippet as final template output
 		// (snippet was used as layout snippet)
-		if (Snippet::$current !== $snippet) {
+		if (
+			Snippet::$current !== null &&
+			Snippet::$current->parent() === $snippet
```
Comment (distantnative) [inline src/Template/Template.php:167]: But isn't `$snippet` just `Snippet::$current`? I can't follow the logic yet

Comment (lukasbestle) [reply src/Template/Template.php:167]: Before the template renders, yes. As soon as the template opens but not closes a layout snippet, the two vars will differ after the template was rendered.

Comment (distantnative) [reply src/Template/Template.php:167]: 👍 maybe you could update the comment as it explains what's happening inside the if statement but not really how this relates to comparing `$snippet` to the parent

Comment (lukasbestle) [reply src/Template/Template.php:167]: ✅

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**